digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/GeometryTransform.scala

```diff
@@ -56,275 +46,65 @@ object RoadSide {
-    val road = validateAndConvertToInt(VkmRoad, data)
-    val roadPart = validateAndConvertToInt(VkmRoadPart, data)
-    val track = validateAndConvertToInt(VkmTrackCode, data)
-    val mValue = validateAndConvertToInt(VkmDistance, data)
-    val deviation = convertToDouble(data.get(VkmOffsetToGivenCoordinates))
-    if (Track.apply(track).eq(Track.Unknown)) {
-      throw new VKMClientException("Invalid value for Track (%s): %d".format(VkmTrackCode, track))
-    }
-    RoadAddress(municipalityCode.map(_.toString), road, roadPart, Track.apply(track), mValue, deviation)
-  }
+    val roadAddress = RoadAddressDAO.getRoadAddress(withLinkIDandMeasure(linkId, mValue.toLong, mValue.toLong, road, includePedestrian)).headOption
 
-  private def mapCoordinates(data: Map[String, Any]) = {
-    val returnCode = data.getOrElse(VkmReturnCode, VkmReturnCodeOk)
-    val error = data.get(VkmError)
-    if (returnCode.toString.toInt == VkmReturnCodeError || error.nonEmpty)
-      throw new VKMClientException("VKM error: %s".format(data.getOrElse(VkmReturnErrorMessage, error.getOrElse(""))))
-    try {
-      val roadAddresses = dig(data, Seq(VkmStartingPoint, VkmQueryRoadAddresses)).asInstanceOf[List[Map[String, Any]]]
-      roadAddresses.map {
-        addr =>
-          val x = dig(addr, Seq("point", "x")).asInstanceOf[Double]
-          val y = dig(addr, Seq("point", "y")).asInstanceOf[Double]
-          Point(x,y)
-      }
-    } catch {
-      case ex: Exception => throw new VKMClientException("Could not convert response from VKM: %s".format(ex.getMessage))
+    val roadSide = roadAddress match {
+      case Some(addrSide) if (addrSide.sideCode.value == assetSideCode) => RoadSide.Right //TowardsDigitizing //
+      case Some(addrSide) if (addrSide.sideCode.value != assetSideCode) => RoadSide.Left //SideCode.AgainstDigitizing //
+      case _ => RoadSide.Unknown
     }
-  }
 
-  private def dig(data: Map[String, Any], keyLine: Seq[String]): Any = {
-    if (keyLine.tail.nonEmpty)
-      dig(data.get(keyLine.head).get.asInstanceOf[Map[String, Any]], keyLine.tail)
-    else
-      data.get(keyLine.head).get
-  }
-
-  private def validateAndConvertToInt(fieldName: String, map: Map[String, Any]) = {
-    def value = map.get(fieldName)
-    if (value.isEmpty) {
-      throw new VKMClientException(
-        "Missing mandatory field in response: %s".format(
-          fieldName))
-    }
-    try {
-      value.get.toString.toInt
-    } catch {
-      case e: NumberFormatException =>
-        throw new VKMClientException("Invalid value in response: %s, Int expected, got '%s'".format(fieldName, value.get))
+    val address = roadAddress match {
+      case Some(addr) if (addr.track.value == assetSideCode || addr.track == Track.Combined) => RoadAddress(Some(municipalityCode.toString), addr.roadNumber.toInt, addr.roadPartNumber.toInt, addr.track, (addr.startAddrMValue + (mValue - addr.startMValue)).toInt, None)
```

Comment:
Assuming that two road address can not be equal, with different trac code to the same link id. I belive that the following condition can be removed.
```
if (addr.track.value == assetSideCode || addr.track == Track.Combined)
```